exporters/metrics/src/main/java/com.google.cloud.opentelemetry.metric/MetricTranslator.java

```diff
@@ -0,0 +1,213 @@
+import com.google.common.primitives.Longs;
+import com.google.monitoring.v3.Point;
+import com.google.monitoring.v3.Point.Builder;
+import com.google.monitoring.v3.TimeInterval;
+import com.google.monitoring.v3.TypedValue;
+import com.google.protobuf.Timestamp;
+import io.opentelemetry.common.AttributeValue;
+import io.opentelemetry.common.ReadableAttributes;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.data.MetricData.Descriptor.Type;
+import io.opentelemetry.sdk.resources.Resource;
+import java.time.Instant;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+public class MetricTranslator {
+
+  private static final Logger logger = Logger.getLogger(MetricTranslator.class.getName());
+
+  static final String DESCRIPTOR_TYPE_URL = "custom.googleapis.com/OpenTelemetry/";
+  static final String UNIQUE_IDENTIFIER_KEY = "opentelemetry_id";
+  static final long NANO_PER_SECOND = (long) 1e9;
+
+  static final Set<Type> GAUGE_TYPES = ImmutableSet.of(NON_MONOTONIC_LONG, NON_MONOTONIC_DOUBLE);
+  static final Set<MetricData.Descriptor.Type> CUMULATIVE_TYPES = ImmutableSet
+      .of(MONOTONIC_LONG, MONOTONIC_DOUBLE);
+  static final Set<MetricData.Descriptor.Type> LONG_TYPES = ImmutableSet.of(NON_MONOTONIC_LONG, MONOTONIC_LONG);
+  static final Set<MetricData.Descriptor.Type> DOUBLE_TYPES = ImmutableSet
+      .of(NON_MONOTONIC_DOUBLE, MONOTONIC_DOUBLE);
+
+  static final Map<String, Map<String, String>> OTEL_TO_GCP_LABELS = ImmutableMap.<String, Map<String, String>>builder()
+      .put("gce_instance", ImmutableMap.<String, String>builder()
+          .put("host.id", "instance_id")
+          .put("cloud.account.id", "project_id")
+          .put("cloud.zone", "zone")
+          .build())
+      .put("gke_container", ImmutableMap.<String, String>builder()
+          .put("k8s.cluster.name", "cluster_name")
+          .put("k8s.namespace.name", "namespace_id")
+          .put("k8s.pod.name", "pod_id")
+          .put("host.id", "instance_id")
+          .put("container.name", "container_name")
+          .put("cloud.account.id", "project_id")
+          .put("cloud.zone", "zone")
+          .build())
+      .build();
+
+
+  static Metric mapMetric(MetricData metric, String type, String uniqueIdentifier) {
+    Metric.Builder metricBuilder = Metric.newBuilder().setType(type);
+    metric.getDescriptor().getConstantLabels().forEach(metricBuilder::putLabels);
+    if (uniqueIdentifier != null) {
+      metricBuilder.putLabels(UNIQUE_IDENTIFIER_KEY, uniqueIdentifier);
+    }
+    return metricBuilder.build();
+  }
+
+  static MetricDescriptor mapMetricDescriptor(MetricData metric, String uniqueIdentifier) {
+    String instrumentName = metric.getInstrumentationLibraryInfo().getName();
+    MetricDescriptor.Builder builder = MetricDescriptor.newBuilder().setDisplayName(instrumentName)
+        .setType(DESCRIPTOR_TYPE_URL + instrumentName);
```

Comment:
This adds the `custom.googleapis.com/OpenTelemetry/` prefix to all metric names.
This is a fine default behavior, but I think we should check if the metric is already prefixed.
e.g. if a metric is called `kubernetes.io/container/uptime` (https://cloud.google.com/monitoring/api/metrics_kubernetes), we should leave it as is.

This is how OpenCensus Go exporter is already implemented: https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver/blob/59d068f8d8ff5b653916aa30cdc4e13c7f15d56e/metrics_proto.go#L419
with a hardcode list of known domains: https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver/blob/59d068f8d8ff5b653916aa30cdc4e13c7f15d56e/metrics_proto.go#L48